src/main/java/com/google/devtools/build/lib/skyframe/SkyframeExecutor.java

```diff
@@ -972,6 +972,12 @@ public boolean storeEvents() {
 
         @Override
         public boolean shouldPropagate(SkyKey depKey, SkyKey primaryKey) {
+          // Bzlmod events do not need to be propagated to analysis nodes. They are propagated to
+          // the node for the main repo mapping, which is unconditionally requested in a separate
+          // Skyframe evaluation for every build, thus ensuring correct incremental behavior.
+          if (isBzlmodKey(depKey)) {
```

Comment:
To be extra safe wrt performance, could you separate out a separate `EventFilter`, say `EVENT_FILTER_WITH_BZLMOD` that delegates to the existing `DEFAULT_EVENT_FILTER_WITH_ACTIONS`? Then we can find a signal (nulness of `repositoryHelpersHolder`?) and choose which `EventFilter` to pass to the evaluator.

Reply:
Done, I use (and made visible) an existing method that was already used as an "is this Bazel" feature flag.